tl-its-umich-edu/instructor-productivity-tools
# I349 react upgrade

I have updated to most of the package to recent except of couple

`@tanstack/react-query`: This version latest is version 5 and it has some change which need adding UseEffect which I want to avoid, but instead I upgraded to V4 that itself had major changes. The repo Owner changed npm package. V5 will be addressed in next release
https://tanstack.com/query/v4/docs/framework/react/guides/migrating-to-react-query-4

ESLint and its supported packages are one version behind the latest release. I chose not to update this since Eslint V9 has breaking changes `eslintrc.js` (eslint configuration) is changed to [eslint.config.js ] (https://eslint.org/docs/latest/use/configure/configuration-files) and the way it is configured within. So felt we can wait. 
https://eslint.org/version-support/

Material UI is V6, opted to stay on V5. Since most of our app still on V5

Use the cli command to check what is latest
1. ncu
2. npm show eslint@8 version

Testplan:
1. Do a docker clean and start the tool.
3. Make sure the Get/PUT function while adding and removing the tool works. Since all those API call from FE is handled by this @tanstack/react-query upgrade
4. Make sure No linting errors in the VSCode.
